.gitlab-ci.yml

```diff
@@ -1,182 +1,206 @@
 
-pup4_8-unit:
-  stage: unit
-  tags:
-    - docker
-  image: ruby:2.1
+.pup_5_5: &pup_5_5
+  image: 'ruby:2.4'
   variables:
-    PUPPET_VERSION: '~> 4.8.0'
-  <<: *cache_bundler
-  <<: *setup_bundler_env
-  <<: *spec_tests
+    PUPPET_VERSION: '~> 5.5.6'
+    BEAKER_PUPPET_COLLECTION: 'puppet5'
+    MATRIX_RUBY_VERSION: '2.4'
 
+.pup_5_latest: &pup_5_latest
+  image: 'ruby:2.4'
+  variables:
+    PUPPET_VERSION: '~> 5.0'
+    BEAKER_PUPPET_COLLECTION: 'puppet5'
+    MATRIX_RUBY_VERSION: '2.4'
 
-# Puppet 4.10 for PE 2017.2 support (EOL:2018-02-21)
-# See: https://puppet.com/misc/puppet-enterprise-lifecycle
+# jobs
 # --------------------------------------
-pup4_10-validation:
-  stage: validation
-  tags:
-    - docker
-  image: ruby:2.1
-  variables:
-    PUPPET_VERSION: '~> 4.10.0'
-  <<: *cache_bundler
+.lint_tests: &lint_tests
+  stage: 'validation'
+  tags: ['docker']
   <<: *setup_bundler_env
-  <<: *validation_checks
+  script:
+    - 'bundle exec rake syntax'
+    - 'bundle exec rake lint'
 
-pup4_10-unit:
-  stage: unit
-  tags:
-    - docker
-  image: ruby:2.1
-  variables:
-    PUPPET_VERSION: '~> 4.10.0'
-  <<: *cache_bundler
+.unit_tests: &unit_tests
+  stage: 'validation'
+  tags: ['docker']
   <<: *setup_bundler_env
-  <<: *spec_tests
-
+  script:
+    - 'bundle exec rake spec'
 
-# Puppet 5.3 for PE 2017.3 support (EOL: 2018-07)
-# See: https://puppet.com/misc/puppet-enterprise-lifecycle
-# --------------------------------------
-pup5_3-validation:
-  stage: validation
-  tags:
-    - docker
-  image: ruby:2.4
-  variables:
-    PUPPET_VERSION: '~> 5.3.0'
-  <<: *cache_bundler
+.compliance_base: &compliance_base
+  stage: 'compliance'
+  tags: ['beaker']
   <<: *setup_bundler_env
-  <<: *validation_checks
 
-pup5_3-unit:
-  stage: unit
-  tags:
-    - docker
-  image: ruby:2.4
-  variables:
-    PUPPET_VERSION: '~> 5.3.0'
-  <<: *cache_bundler
+.acceptance_base: &acceptance_base
+  stage: 'acceptance'
+  tags: ['beaker']
   <<: *setup_bundler_env
-  <<: *spec_tests
 
+# Pipeline / testing matrix
+#=======================================================================
 
-# Keep an eye on the latest puppet 5
-# ----------------------------------
-pup5_latest-validation:
-  stage: validation
-  tags:
-    - docker
-  image: ruby:2.4
-  variables:
-    PUPPET_VERSION: '~> 5.0'
-  <<: *cache_bundler
+sanity_checks:
+  <<: *pup_4_latest
   <<: *setup_bundler_env
-  <<: *validation_checks
-
-pup5_latest-unit:
-  stage: unit
-  tags:
-    - docker
-  image: ruby:2.4
-  variables:
-    PUPPET_VERSION: '~> 5.0'
-  <<: *cache_bundler
-  <<: *setup_bundler_env
-  <<: *spec_tests
+  stage: 'sanity'
+  tags: ['docker']
+  script:
+    - 'bundle exec rake check:dot_underscore'
+    - 'bundle exec rake check:test_file'
+    - 'bundle exec rake pkg:check_version'
+    - 'bundle exec rake pkg:compare_latest_tag'
+
+pup4-lint:
+  <<: *pup_4_latest
+  <<: *lint_tests
+
+pup5-lint:
+  <<: *pup_5_latest
+  <<: *lint_tests
+
+pup4.10-u:
```

Comment:
Would rather see puppet abbreviated to 'p' than unit abbreviated to 'u'

Reply:
Why 'p'?

Reply:
p4.10-unit  because unit is more important that puppet